pydantic/json_schema.py

```diff
@@ -2427,9 +2445,34 @@ def __get_pydantic_json_schema__(
         json_schema = handler(core_schema)
         if mode != handler.mode:
             return json_schema
-        examples = json_schema.get('examples', {})
-        examples.update(to_jsonable_python(self.examples))
-        json_schema['examples'] = examples
+        examples = json_schema.get('examples')
+        if examples is None:
+            json_schema['examples'] = to_jsonable_python(self.examples)
+        if isinstance(examples, dict):
+            if isinstance(self.examples, list):
+                warnings.warn(
+                    'Updating existing JSON Schema examples of type dict with examples of type list. '
+                    'Only the existing examples values will be retained.',
+                    UserWarning,
+                )
```

Comment:
Let's warn here (and on the next one) that `dict` is deprecated as well - I realize this will be raised in a separate warning, but I think it makes sense to reiterate in this one.